models/BaseEntity.cfc

```diff
@@ -63,11 +77,13 @@ component accessors="true" {
     }
 
     function getAttributesData( aliased = false, withoutKey = false ) {
+
         getAttributes().keyArray().each( function( key ) {
-            if ( variables.keyExists( key ) && ! isReadOnlyAttribute( key ) ) {
-                setAttribute( key, variables[ key ] );
+            if ( ! isReadOnlyAttribute( key ) ) {
+              setAttribute( key, !isNull( variables[ key ] ) ? variables[ key ] : ''  );
```

Comment:
I'm not sure I want to set `null` variables to empty strings.  I think I'd like to keep them null, especially with full null support on the horizon.

Reply:
That's fair, but then you're not going to have a representation of those keys in your memento in all existing engines ( and the ones with null support turned off ).  I would rather have an option to turn on full null support and default to having the key present.

Reply:
That's been requested elsewhere, as well.  I think we need a module setting for this.  Probably outside the scope of this PR.

Reply:
At least, for now, I believe we should have a full representation of the memento.  Otherwise you need a default for every property, which may or may not represent the database return typing and value.

Reply:
If full null was turned off, wouldn't the `queryExecute` call return an empty string anyway?

Reply:
After execution, yes. The problem is the state of the memento in a newly created entity.  There's no visibility in to the persistent columns through the memento or the attributes, because null is the same as it not existing.

You can say as a developer "I know this is a property", for of course, but this ensures that the key has visiblility.

Reply:
We should have visibility into the persisted columns because we have an array of the persistent attributes available in `getAttributes`.  Without `legacyNullSupport` it would return something like:

Reply:
The attributes are set from the metadata introspection.  They don't get set if the property is null, which is the reason for this pull request.  

When `getMemento()` returns `getAttributes` the null variables no longer exist.  What is `undefined`?  That's not a CF thing.  In an engine that doesn't have full null, you have no way to check for the available keys if it's a null because `structKeyArray()` and `for( var key in getMemento()` are going to exclude the nulls.

In any case, we shouldn't write a new module targeted at full null support, at this point. It's fine to be a bit opinionated and I wholly agree on trying to develop toward the future, but you have over 20+ years of null support and a half-dozen engine versions that have treated nulls as non-existent.  We need to develop with that in mind, IMHO.